src/authorization/WebAclAuthorizer.ts

```diff
@@ -66,6 +69,8 @@ export class WebAclAuthorizer extends Authorizer {
     const modeString = ACL[this.capitalize(mode) as 'Write' | 'Read' | 'Append' | 'Control'];
     const auths = store.getQuads(null, ACL.mode, modeString, null).map((quad: Quad): Term => quad.subject);
     if (!auths.some((term): boolean => this.hasAccess(agent, term, store))) {
+      this.logger.warn(`No permission to use ${mode} mode (agent not 
+      ${typeof agent.webID === 'string' ? 'allowed to access data' : 'authorized'}).`);
       throw typeof agent.webID === 'string' ? new ForbiddenHttpError() : new UnauthorizedHttpError();
```

Comment:
```suggestion
    if (typeof agent.webID === 'string') {
      this.logger.warn(`Agent #{agent.webID} unauthorized for ${mode}.`);
    }
      this.logger.warn(`No permission to use ${mode} mode (agent not 
      ${typeof agent.webID === 'string' ? 'allowed to access data' : 'authorized'}).`);
      throw typeof agent.webID === 'string' ? new ForbiddenHttpError() : new UnauthorizedHttpError();
```

Plus document `getAclRecursive` to include something like

```javascript
     this.logger.info(`Obtaining ACL for #{identifier}`);
```

such that we see what ACL is being checked.

Reply:
Do you want the 2 indicated lines to stay as they are? (So that if ` typeof agent.webID === 'string'` 2 similar warnings are logged.)

For ACL: do you want a log for every try or only for the latest successful recursive step?

Reply:
Those are fine for me indeed; perhaps add a `const isLoggedIn = typeof agent.webID === 'string'` to avoid repetition.

Reply:
ACL: every try with `debug`, successful with `info`.